libraries/APM_Control/AR_AttitudeControl.cpp

```diff
@@ -330,19 +334,19 @@ float AR_AttitudeControl::get_throttle_out_speed(float desired_speed, bool motor
         return 0.0f;
     }
 
-    // calculate dt
+    // if not called recently, reset input filter and desired speed to actual speed (used for accel limiting)
     const uint32_t now = AP_HAL::millis();
-    float dt = (now - _speed_last_ms) / 1000.0f;
-    if ((_speed_last_ms == 0) || (dt > (AR_ATTCONTROL_TIMEOUT_MS / 1000.0f))) {
-        dt = 0.0f;
+    if ((_speed_last_ms == 0) || ((now - _speed_last_ms) > AR_ATTCONTROL_TIMEOUT_MS)) {
         _throttle_speed_pid.reset_filter();
-    } else {
-        _throttle_speed_pid.set_dt(dt);
+        _desired_speed = speed;
```

Comment:
Is this supposed to be `desired_speed` (without the underscore prefix)? The reason I'm asking is that this is overwritten right below so this assignment doesn't seem to do anything.

Reply:
Thanks very much for the review, it helps enormously.

I think setting the _desired_speed is correct.  We use the _desired_speed (with the underscore) as the basis for the accel limiting.  Normally this variable is the desired speed from the previous iteration but in this case we are setting it to the vehicle's actual speed.